src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -14,120 +14,103 @@
 	}
 	else
 	{
-		randomcoeff_additive = NULL;
+		m_randomoffset = NULL;
 	}
 
-	if((cur_dim_feature_space>0)&&(cur_dim_input_space>0))
+	if((cur_m_dim_feature_space>0)&&(cur_dim_input_space>0))
 	{
-		if(feats.randomcoeff_multiplicative==NULL)
+		if(feats.m_randomweights==NULL)
 		{
 			throw ShogunException(
-							"void CRandomFourierGaussPreproc::copy(...): feats.randomcoeff_multiplicative==NULL && cur_dim_feature_space>0 &&(cur_dim_input_space>0)  \n");
+							"void CRandomFourierGaussPreproc::copy(...): feats.m_randomweights==NULL && cur_m_dim_feature_space>0 &&(cur_dim_input_space>0)  \n");
 		}
 
-		randomcoeff_multiplicative=SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-		std::copy(feats.randomcoeff_multiplicative,feats.randomcoeff_multiplicative+cur_dim_feature_space*cur_dim_input_space,randomcoeff_multiplicative);
+		m_randomweights=SG_MALLOC(float64_t, cur_m_dim_feature_space*cur_dim_input_space);
+		std::copy(feats.m_randomweights,feats.m_randomweights+cur_m_dim_feature_space*cur_dim_input_space,m_randomweights);
 	}
 	else
 	{
-		randomcoeff_multiplicative = NULL;
+		m_randomweights = NULL;
 	}
 
 }
 
-RandomFourierGaussPreproc::RandomFourierGaussPreproc() :
-	RandomMixin<DensePreprocessor<float64_t>> () {
-	dim_feature_space = 1000;
-	dim_input_space = 0;
-	cur_dim_input_space = 0;
-	cur_dim_feature_space=0;
-
-	randomcoeff_multiplicative=NULL;
-	randomcoeff_additive=NULL;
-
-	kernelwidth=1;
-	cur_kernelwidth=kernelwidth;
-
-	SG_ADD(&dim_input_space, "dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&cur_dim_input_space, "cur_dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&dim_feature_space, "dim_feature_space",
-			"Dimensionality of the feature space.");
-	SG_ADD(&cur_dim_feature_space, "cur_dim_feature_space",
-			"Dimensionality of the feature space.");
+RandomFourierGaussPreproc::RandomFourierGaussPreproc()
+    : RandomMixin<DensePreprocessor<float64_t>>()
+{
+	m_dim_feature_space = 1000;
 
-	SG_ADD(&kernelwidth, "kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-	SG_ADD(&cur_kernelwidth, "cur_kernelwidth", "Kernel width.", ParameterProperties::HYPER);
+	m_randomweights = NULL;
+	m_randomoffset = NULL;
 
-	watch_param(
-		"randomcoeff_additive", &randomcoeff_additive,
-		&cur_dim_feature_space);
+	m_kernelwidth = 1;
 
-	watch_param(
-		"randomcoeff_multiplicative", &randomcoeff_multiplicative,
-		&cur_dim_feature_space, &cur_dim_input_space);
+	SG_ADD(
+	    &m_dim_feature_space, "m_dim_feature_space",
+	    "Dimensionality of the feature space.");
+	SG_ADD(
+	    &m_kernelwidth, "m_kernelwidth", "Kernel width.",
+	    ParameterProperties::HYPER);
+
+	
+	watch_param("m_randomoffset", &m_randomoffset, &cur_m_dim_feature_space);
 
+	watch_param(
+	    "m_randomweights", &m_randomweights, &cur_m_dim_feature_space,
+	    &cur_dim_input_space);//TODO: what to put here? :/
```

Comment:
Can i have some help here? @karlnapf

Reply:
you want to make this an SGMatrix and the just register it as SGMatrices are registered elsewhere. See eg DenseFeatures